File: apps/web/src/features/catalog/catalog-category-images.test.ts
import { describe, expect, it } from 'vitest';

import { catalogCategoryImageSource } from './catalog-category-images';

describe('catalogCategoryImageSource', () => {
  it('maps a system category to its catalog photograph', () => {
    expect(catalogCategoryImageSource('aceite, especias y salsas')).toBe('/catalog-images/catalog_aceite_especias_y_salsas.webp');
  });

  it('leaves custom categories without a photograph on the icon fallback', () => {
    expect(catalogCategoryImageSource('productos de casa')).toBeNull();
  });
});

File: apps/android/feature/shoppinglist/src/test/java/dev/esgarpe/nfcompra/feature/shoppinglist/CatalogCategoryImagesTest.kt
package dev.esgarpe.nfcompra.feature.shoppinglist

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test

class CatalogCategoryImagesTest {
    @Test
    fun `maps a system category to its catalog photograph`() {
        assertEquals(
            R.drawable.catalog_aceite_especias_y_salsas,
            catalogCategoryImageRes("aceite, especias y salsas"),
        )
    }

    @Test
    fun `leaves custom categories without a photograph on the icon fallback`() {
        assertNull(catalogCategoryImageRes("productos de casa"))
    }
}
